core/field_textinput.js

```diff
@@ -128,6 +135,50 @@ Blockly.FieldTextInput.prototype.configure_ = function(config) {
   }
 };
 
+/**
+ * Create the block UI for this field.
+ * @package
+ */
+Blockly.FieldTextInput.prototype.initView = function() {
+  var renderer = this.sourceBlock_.workspace.getRenderer();
+  if (renderer.getConstants().FULL_BLOCK_FIELDS) {
+    // Step one: figure out if this is the only field on this block.
```

Comment:
Not necessary, but might be nice to put some of the logic for figuring out if it is a fullBlockClickTarget in its own method.

Reply:
Agreed.  I'm going to see how this works out with the colour field (which may be the only other field that needs this function directly) and then see if I can move all of this logic onto the renderer instead.